dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityAttributeStore.java

```diff
@@ -176,6 +183,57 @@ public HibernateTrackedEntityAttributeStore( SessionFactory sessionFactory, Jdbc
             .collect( Collectors.toSet() );
     }
 
+    @Override
+    public Set<TrackedEntityAttribute> getAllSearchableAndUniqueTrackedEntityAttributes()
+    {
+        Set<TrackedEntityAttribute> result = new HashSet<>();
+
+        Query programTeaQuery = sessionFactory.getCurrentSession()
+            .createQuery( "select attribute from ProgramTrackedEntityAttribute ptea where ptea.searchable=true" );
+        Query tetypeAttributeQuery = sessionFactory.getCurrentSession()
+            .createQuery(
+                "select trackedEntityAttribute from TrackedEntityTypeAttribute teta where teta.searchable=true" );
+        Query uniqueAttributeQuery = sessionFactory.getCurrentSession()
+            .createQuery( "from TrackedEntityAttribute tea where tea.unique=true" );
+
+        List<TrackedEntityAttribute> programSearchableTrackedEntityAttributes = programTeaQuery.list();
+        List<TrackedEntityAttribute> trackedEntityTypeSearchableAttributes = tetypeAttributeQuery.list();
+        List<TrackedEntityAttribute> uniqueAttributes = uniqueAttributeQuery.list();
+
+        result.addAll( programSearchableTrackedEntityAttributes );
+        result.addAll( trackedEntityTypeSearchableAttributes );
+        result.addAll( uniqueAttributes );
+
+        return result;
+    }
+
+    @Override
+    public boolean createTrigramIndexForAttribute( TrackedEntityAttribute trackedEntityAttribute )
+    {
+        String query = trigramIndexQuery( trackedEntityAttribute );
```

Comment:
I would inline the method. The code it hides is so easy to read and matches the abstraction level with the rest of the code in here. Easy to read and less navigation 😋

Reply:
Roger that!